client-js/app/elements/labrad-plot.ts

```diff
@@ -222,6 +247,55 @@ export class Plot extends polymer.Base {
             .attr('y', 0)
             .attr('width', width)
             .attr('height', height);
+
+
+    // Color Bar Axis
+    if (this.numIndeps == 2) {
+      p.zAxis = d3.svg.axis()
+              .scale(p.zScale)
+              .orient('right')
+              .ticks(7)
+              .tickSize(5);
+
+      // The Color Bar Gradient
+      p.svg.append('defs').append("linearGradient")
+          .attr("id", "ColorBarGradient")
+          .attr("x1", "0%")
+          .attr("y1", "0%")
+          .attr("x2", "0%")
+          .attr("y2", "100%")
+          .selectAll("stop")
+          .data([
+            {offset: "0%", color: COLOR_MAP[0]},
+            {offset: "50%", color: COLOR_MAP[128]},
+            {offset: "100%", color: COLOR_MAP[255]}
+          ])
```

Comment:
This is relying on the SVG implementation in the browser to do a uniform linear gradient.

I forgot that Viridis went to special lengths to make the gradient uniform in color. I will sample more points and see how that goes.

The inefficiency you mention is true for everything in `createPlot` that's why for #185 it only gets called once, so I'm not too concerned about this part in particular at this stage.